### ISSUE TITLE
Upgrade to React16 breaks styled components

### ISSUE THREAD

**Issue by Bennyz811** [styled-components/styled-components#2835]
## BUG TEMPLATE
    Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.
    
    * Move code with side effects to componentDidMount, and set initial state in the constructor.
    * Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
    
    Please update the following components: Select__StyledLabel

```
const StyledLabel = styled.label`
   color: ${props => props.theme.color};
   background: ${props => props.theme.background};
   border: ${props => props.theme.borderColor};
   border-radius: ${props => props.theme.borderRadius};

    &:before {
      border: 1px solid ${props => props.theme.primary ? props.theme.primary : '#0b68cf'};
      background-color: '#fff';
      display: 'inline-block';
    }
    &.checked {
      background: ${props => {
        const theme = createTheme(...Object.keys(props.theme));
          return theme.checkedSelectButtonColor ? theme.checkedSelectButtonColor.darken(0.2) : 
        }
      }} !important;

      color: ${props => props.theme.checkedSelectButtonFontColor ? '#000' : '#fff')} !important;

      &:before {
        background-color: ${props => props.theme.primary ? props.theme.primary : '#fff'};
      }
    }
`;
```

## Environment
    "react": "^16.9.0",
    "react-redux": "^6.0.1",
    "react-router": "^5.1.2",
    "react-router-dom": "^5.1.2",
    "styled-components": "^4.0.0",

## Expected Behavior
every styles to be applied correctly

## Actual Behavior
no styles coming through. Colors are not applying as expected

**Comment by CodefulDom** [comment]
@Bennyz811 I am getting the same behavior.

**Comment by probablyup** [comment]
Need some sort of codesandbox reproduction to be able to help you. Many people use v4 with React 16, so it's likely a configuration issue.

**Comment by joseph-d-p** [comment]
I'm getting the same warning using [@react-pdf/renderer](https://github.com/diegomura/react-pdf).

```
const StyledView = styled(View)``;
const StyledText = styled(Text)``;

<Document>
  <Page size="A4">
    <StyledView>
      <StyledText>Hello World</StyledText>
    </StyledView>
  </Page>
</Document>
```

packages:
```
"react": "^16.12.0",
"react-dom": "^16.12.0",
"styled-components": "^5.1.0",
```

**WARNING** (for both `componentWillMount` and `componentWillReceiveProps`)
> react-reconciler.development.js:2027 Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.
> * Move code with side effects to componentDidMount, and set initial state in the constructor.
>* Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
>Please update the following components: styled.TEXT, styled.VIEW

> react-reconciler.development.js:2027 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.
> * Move data fetching code or side effects to componentDidUpdate.
> * If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
> * Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
> Please update the following components: styled.TEXT, styled.VIEW

**Comment by probablyup** [comment]
Use v5

Sent via mobile, please excuse brevity and/or typos.
________________________________
